File: python/nt_solutions/coding_1_approximation/exo8.py
fL = np.zeros([n, n])

for i in range(1,n//w+1):
    for j in range(1,n//w+1):
        fL[(i-1)*w: i*w, (j-1)*w: j*w] = dct2(f[(i-1)*w: i*w, (j-1)*w: j*w])

File: r/nt_solutions/coding_1_approximation/exo8.R




fL <- matrix(rep(0, length=n*n), c(n, n))

for (i in 1:(n%/%w)){
  for (j in 1:(n%/%w)){
    fL[((i-1)*w+1):(i*w), ((j-1)*w+1):(j*w)] = dct2(as.matrix(f)[((i-1)*w+1):(i*w), ((j-1)*w+1):(j*w)])
  }
}
